components/eventing-controller/reconciler/subscription/reconciler_test.go

```diff
@@ -103,50 +103,158 @@ var _ = Describe("Subscription Reconciliation Tests", func() {
+				v1.ConditionFalse,
+				"",
+			)
 			getSubscription(givenSubscription, ctx).Should(And(
 				reconcilertesting.HaveSubscriptionName(subscriptionName),
 				reconcilertesting.HaveCondition(subscriptionAPIReadyFalseCondition),
 			))
 
-			By("Deleting the object to not provoke more reconciliation requests")
-			Expect(k8sClient.Delete(ctx, givenSubscription)).Should(BeNil())
-			getSubscription(givenSubscription, ctx).ShouldNot(reconcilertesting.HaveSubscriptionFinalizer(Finalizer))
+			Context("Fixing an invalid Subscirption should update the APIRule to ready", func() {
+				validSink := fmt.Sprintf("https://%s.%s.svc.cluster.local/path1", subscriberSvc.Name, subscriberSvc.Namespace)
+				givenSubscription.Spec.Sink = validSink
+				updateSubscription(givenSubscription, ctx).Should(reconcilertesting.HaveSubscriptionSink(validSink))
+
+				By("Creating a valid APIRule")
+				getAPIRuleForASvc(subscriberSvc, ctx).Should(reconcilertesting.HaveNotEmptyAPIRule())
+				apiRuleUpdated := filterAPIRulesForASvc(getAPIRules(ctx, subscriberSvc), subscriberSvc)
+				getAPIRule(&apiRuleUpdated, ctx).Should(And(
+					reconcilertesting.HaveNotEmptyHost(),
+					reconcilertesting.HaveNotEmptyAPIRule(),
+					reconcilertesting.HaveAPIRuleOwnersRefs(givenSubscription.UID),
+				))
+
+				By("Updating the APIRule status to be Ready")
+				ensureAPIRuleStatusUpdatedWithStatusReady(&apiRuleUpdated, ctx).Should(BeNil())
+
+				By("Setting a subscription active condition")
+				subscriptionActiveCondition := eventingv1alpha1.MakeCondition(
+					eventingv1alpha1.ConditionSubscriptionActive,
+					eventingv1alpha1.ConditionReasonSubscriptionActive,
+					v1.ConditionTrue,
+					"",
+				)
+				getSubscription(givenSubscription, ctx).Should(And(
+					reconcilertesting.HaveSubscriptionName(subscriptionName),
+					reconcilertesting.HaveCondition(subscriptionActiveCondition),
+				))
+
+				By("Setting APIRule status in Subscription to Ready")
+				subscriptionAPIReadyCondition := eventingv1alpha1.MakeCondition(
+					eventingv1alpha1.ConditionAPIRuleStatus,
+					eventingv1alpha1.ConditionReasonAPIRuleStatusReady,
+					v1.ConditionTrue,
+					"",
+				)
+				getSubscription(givenSubscription, ctx).Should(And(
+					reconcilertesting.HaveSubscriptionName(subscriptionName),
+					reconcilertesting.HaveAPIRuleName(apiRuleUpdated.Name),
+					reconcilertesting.HaveCondition(subscriptionAPIReadyCondition),
+					reconcilertesting.HaveSubscriptionReady(),
+				))
+
+				By("Sending valid webhook requests only")
+				validRequests, invalidRequests := countRequestsForWebhooks(reconcilertesting.EventType, reconcilertesting.EventTypeNotClean)
+				Expect(validRequests).Should(reconcilertesting.BeGreaterThanOrEqual(1))
+				Expect(invalidRequests).Should(BeZero())
+			})
+		})
+	})
+
+	When("Two Subscriptions which are using the same sink but different paths are deleted", func() {
+		It("Should update the APIRule accordingly", func() {
```

Comment:
This test is doing a lot of things. It is critical to group the steps based on a common theme. Can we do something like:
```
When "Two Subscriptions using different sinks are made to use the same sink and then both are deleted"
  It "Should create 2 subscriptions with different sinks"
  It "Should delete the first subscription and change the APIRule"
  It "Should delete the second subscription and delete APIRule"
```

Reply:
Refactoring the test should be tackled in this task: #10781